deps-injector/deps-injector.js

```diff
@@ -6,57 +6,85 @@ require.config({
+                    PumaInjector.CONFIGURATION === 'TEST' ? that.pumaTest(url, this.responseText, that) : that.pumaInjector(url, this.responseText, that);
                 }
                 catch (e) {
                     console.log('Error in puma interpretation');
-                    console.error(e);
-                    if(PumaInjector.FORCE_INJECT) that.createScriptNode(dependency);
+                    if (PumaInjector.FORCE_INJECT) that.createScriptNode(url);
                 }
             }
         };
-        xhttp.open("GET", dependency.url, true);
+        xhttp.open("GET", url, true);
         xhttp.send();
     };
 
+    PumaInjector.prototype.pumaTest = function(url, responseText, that) {
+        console.info('********** Entering ', url, '****************************');
+        var result = that.puma.evalPuma(responseText, 'inyector');
+
+        if (result.success !== undefined) {
+            if (result.success) {
+                console.info('++++++++++++++ Successful injection ++++++++++++++++++');
+                that.createScriptNode(url);
+            } else {
+                console.error(`Error when interpreting the file, puma does not support any internal components.
+                The error occurred in the line: ${result.pumaAst.loc.end.line}, column: ${result.pumaAst.loc.end.column}
+                The component that generates error is the following: ${result.output}`);
+                if (PumaInjector.FORCE_INJECT) that.createScriptNode(url);
+            }
+        }
+        else {
+            console.error(`Error when interpreting the file, puma does not support any internal components.
+            The error occurred in the line: ${result.loc.end.line}, column: ${result.loc.end.column}
+            The component that generates error is the following: ${result.name}`);
+        }
+        console.info('*************************END INJECTION**********************************');
+    }
+
+    PumaInjector.prototype.pumaInjector = function(url, responseText, that) {
+        var result = that.puma.evalPuma(responseText);
+        that.createScriptNode(url);
+    }
+
 
-    PumaInjector.prototype.createScriptNode = function(dependency){
+    PumaInjector.prototype.createScriptNode = function (url) {
         var script = document.createElement('script');
-        script.src = dependency.url;
-        script.setAttribute("crossorigin", "anonymous");
-        script.integrity = dependency.checksum;
+        script.src = url;
         document.getElementsByTagName('head')[0].appendChild(script);
     };
 
-    PumaInjector.prototype.installDeps = function(deps){
+    PumaInjector.prototype.installDeps = function (deps) {
         var dependencyList = deps || this.deps;
 
         //Retrieve the script from CDN's
-        for(var i=0;i<dependencyList.length;i++){
-            this.getExternalScript(dependencyList[i]);
+        for (var key in dependencyList) {
```

Comment:
Style C for is more performant in JS.

Reply:
Well, I make puma.js an array and not an object.